src/VisualStudio/Core/Def/Implementation/Watson/WatsonReporter.cs

```diff
@@ -93,17 +93,16 @@ public static void ReportNonFatal(Exception exception)
                 exceptionObject: exception,
                 gatherEventDetails: faultUtility =>
                 {
-                    // add current process dump
-                    faultUtility.AddProcessDump(currentProcess.Id);
-
                     // add ServiceHub log files:
                     foreach (var path in CollectServiceHubLogFilePaths())
                     {
                         faultUtility.AddFile(path);
                     }
 
                     // Returning "0" signals that we should send data to Watson; any other value will cancel the Watson report.
-                    return 0;
+                    // We never want to trigger watson manually, we'll let TargetNotifications determine if a Watson should
+                    // be reported. See https://aka.ms/roslynnfwdocs for more details
+                    return 1;
```

Comment:
After looking at the FaultEvent code, I need to correct what I said earlier. You want to always return 0 from this function. The Watson report will still be sampled.

Also for the comment, it would probably be most correct to say something like this:

Returning "0" signals that, if sampled, we should send data to Watson. Any other value will cancel the Watson report. We never want to trigger a process dump manually, we'll let TargetedNotifications determine if a dump should be collected.

Reply:
Thanks Michael. I changed back to returning 0 and updated the comment.